### PR TITLE
Automatic APIs versions from GitHub tags

### PR DESCRIPTION
**Pull request by juanignaciosl**
Comes from a leapfrog. It avoids hardcoding APIs versions.

### CODE REVIEW

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -261,4 +262,22 @@ def safe_json_get(url)
...
+    @sql_api_version = last_version(JSON.parse(http_client.get("#{REPOS}/CartoDB-SQL-API/tags").body))
+  end
+
+  def major_minor_version(version)
+    match = VERSION_REGEXP.match(version)
+    "#{match[1]}.#{match[2]}"
+  end
+
+  def last_version(versions)
+    versions.map { |v| v['name'] }.map { |vn| VERSION_REGEXP.match(vn) }.compact.map { |v| v[0] }.first
```
Comment (javitonino) [inline app/controllers/home_controller.rb]: You could sort this list before taking the `first` element, as this depends on the order of GitHub response.

File: app/controllers/home_controller.rb [hunk 2]
```diff
@@ -261,4 +262,22 @@ def safe_json_get(url)
     { 'error fetching info' => e.message }
   end
 
+  GITHUB_API = "https://api.github.com".freeze
+  REPOS = "#{GITHUB_API}/repos/cartodb".freeze
+  VERSION_REGEXP = /([^.]+)\.([^.]+)\.[^.]/
```
[the remark this answers is not in the document]
Comment (juanignaciosl) [reply app/controllers/home_controller.rb]: Oh, in fact I searched for it (I couldn't remember the class name) and didn't found it, thanks!

File: app/controllers/home_controller.rb (hunk 1 above)
Comment (juanignaciosl) [reply app/controllers/home_controller.rb]: Github response is ordered, I opted to trust them. I'll sort, anyway.

**Comment by juanignaciosl** [comment]
> if you are running an old version of CARTO, the diagnosis page will always tell you to upgrade

🤔  I hadn't thought about this. My use case is being up-to-date, but it's true that this might be even worse. Currently, we haven't got the explicit dependency. Maybe I should just let this go :-\

I'll give this a thought. We should probably have a dependency table, but it's not something that should be decided because of a leapfrog.

**Comment by juanignaciosl** [comment]
Closing this, as it should probably be replaced by using `carto-package.json`.

**Pull request closed without merging**